### PR TITLE
slides: init at 0.2.0

### PR DESCRIPTION
**Pull request by maaslalani**
###### Motivation for this change

Packages https://github.com/maaslalani/slides: A terminal based presentation tool.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/slides/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  doCheck = false;
+
+  buildFlagsArray = [ "-ldflags= -s -w -X=main.Version=${version}" ];
+
+  meta = with lib; {
+    description = "Terminal based presentation tool";
+    homepage = "https://github.com/maaslalani/slides";
+    license = licenses.mit;
+    maintainers = [];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slides/default.nix:24]: That needs to be set.

File: pkgs/applications/misc/slides/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "maaslalani";
+    repo = "slides";
+    rev = "v${version}";
+    sha256 = "0gsjq8866rdbhgl8nmawx3iqm4ww26xlnybwl16a4mwvwcai0min";
+  };
+
+  vendorSha256 = "1wld36xi7ybl8k6m9kvszhrzr492k5yl87p5vc8vkx21bnblv1qp";
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slides/default.nix:16]: Please add a comment why tests are disabled.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/slides/default.nix (hunk 1 above)
Comment (maaslalani) [reply pkgs/applications/misc/slides/default.nix]: 👍 set

File: pkgs/applications/misc/slides/default.nix (hunk 2 above)
Comment (maaslalani) [reply pkgs/applications/misc/slides/default.nix]: Sorry, I don't think there was a reason. I enabled it, thanks!

File: pkgs/applications/misc/slides/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "maaslalani";
+    repo = "slides";
+    rev = "v${version}";
+    sha256 = "0gsjq8866rdbhgl8nmawx3iqm4ww26xlnybwl16a4mwvwcai0min";
+  };
+
+  vendorSha256 = "1wld36xi7ybl8k6m9kvszhrzr492k5yl87p5vc8vkx21bnblv1qp";
+
+  doCheck = true;
```
Comment (fabaff) [inline pkgs/applications/misc/slides/default.nix:16]: Not needed, it's a default.

File: pkgs/applications/misc/slides/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  buildFlagsArray = [ "-ldflags= -s -w -X=main.Version=${version}" ];
```
Comment (fabaff) [inline pkgs/applications/misc/slides/default.nix:18]: You could use `ldflags = [];`. Was added recently (https://github.com/NixOS/nixpkgs/pull/125520).

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -6140,6 +6140,12 @@
     githubId = 36235154;
     name = "Sean Haugh";
   };
+  maaslalani = {
```
Comment (fabaff) [inline maintainers/maintainer-list.nix:6143]: Please move this change to a separate commit in this PR.

File: pkgs/applications/misc/slides/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "slides";
+  version = "0.1.1";
```
Comment (fabaff) [inline pkgs/applications/misc/slides/default.nix:5]: [0.1.2](https://github.com/maaslalani/slides/releases/tag/v0.1.2) seems to be the latest release.

File: pkgs/applications/misc/slides/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/slides/default.nix:18]: You didn't push that change.

Comment (maaslalani) [reply pkgs/applications/misc/slides/default.nix:18]: When I bumped to `0.2.0` then I did need the checks to be disabled so I added a comment.

Comment (SuperSandro2000) [reply pkgs/applications/misc/slides/default.nix:18]: One last question: do we have the program that are required for the tests already in nixpkgs? You can add them with checkInputs. I think we should try enabling tests to make sure the package works correctly.

**Comment by ethancedwards8** [comment]
Result of `nixpkgs-review pr 126299` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>slides</li>
  </ul>
</details>

File: pkgs/applications/misc/slides/default.nix (hunk 2 above)
Comment (maaslalani) [reply pkgs/applications/misc/slides/default.nix]: Sorry about the late response, yes the programs are in `nixpkgs` my tests use bash, go, ruby, and python to verify that the different languages code can be executed if they are code blocks on the slides (I will probably remove these tests in a later version however, I can update this package when I remove those tests).

Comment (maaslalani) [reply pkgs/applications/misc/slides/default.nix]: I updated the `checkInputs` to match what is expected on the tests for now and turned the checks back on.

File: pkgs/applications/misc/slides/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    owner = "maaslalani";
+    repo = "slides";
+    rev = "v${version}";
+    sha256 = "0vwpisyvsgmvrzp26hapghgqmm3fhrzrvdnmws00xc2q3fsij9qx";
+  };
+
+  checkInputs = [
+    bash
+    go
+    python39
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slides/default.nix:17]:
```suggestion
    python3
```

File: pkgs/applications/misc/slides/default.nix [hunk 8]
```diff
@@ -0,0 +1,34 @@
+{ lib, buildGoModule, fetchFromGitHub, bash, go, python39, ruby }:
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/slides/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, bash, go, python3, ruby }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/slides/default.nix (hunk 7 above)
Comment (maaslalani) [reply pkgs/applications/misc/slides/default.nix]: Fixed

File: pkgs/applications/misc/slides/default.nix (hunk 8 above)
Comment (maaslalani) [reply pkgs/applications/misc/slides/default.nix]: Fixed
